experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/proxies.ts

```diff
@@ -67,10 +67,28 @@ export function getProxyForField<TSchema extends FieldSchema>(
 			return getProxyForNode(asValue.boxedContent) as ProxyField<TSchema>;
 		}
 		case FieldKinds.optional: {
-			fail(`"not implemented"`);
+			const asValue = field as TypedField<FieldSchema<typeof FieldKinds.optional>>;
+
+			// TODO: Ideally, we would return leaves without first boxing them.  However, this is not
+			//       as simple as calling '.content' since this skips the node and returns the FieldNode's
+			//       inner field.
+
+			// TODO: Remove non-null assertion if/when ET2 editing API becomes internal.
+			//
+			//       The non-null assertion is required because the ET2 API forces undefined optional fields
+			//       into existence when setting maps via getting the key for the non-existent field.
+			//
+			//       Normally, undefined fields are elided in enumeration and therefore do not expect to
+			//       encounter them in the proxy-based API.
+
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			return getProxyForNode(asValue.boxedContent!) as ProxyField<TSchema>;
```

Comment:
Just double checking that this is right. I can see how we don't have to worry about acquiring empty optional fields from maps, because we won't hand out empty fields from map lookups in the proxy API. And when getting optional fields from an object, the `tryGetField` function skips the field entirely if it's empty. What about the root of the tree being an optional field, and us starting with that? Is that not a problem because we're going to require the root field to be a "Required"?

Reply:
I forgot about the special root field.  Good catch.